pkg/kubectl/cmd/login.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	utilflag "k8s.io/kubernetes/pkg/util/flag"
+)
+
+func NewCmdLogin(f *cmdutil.Factory, pathOptions *clientcmd.PathOptions, out io.Writer) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "login Log in to the cluster.",
+		Long:  `login obtains the neccesary credentials to make authenticated requests to the cluster. The type of log in depends on the 'auth-provider' type in your configuration, and will likely require some type of user interaction.`,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunLogin(out, cmd, f, pathOptions, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	return cmd
+}
+
+func RunLogin(out io.Writer, cmd *cobra.Command, f *cmdutil.Factory, pathOptions *clientcmd.PathOptions, args []string) error {
+
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	flags.SetNormalizeFunc(utilflag.WarnWordSepNormalizeFunc) // Warn for "_" flags
+
+	cmdCliCfg := cmdutil.DefaultClientConfig(flags)
```

Comment:
> Perhaps this is a desirable feature, but as a first pass, is it not reasonable to simply output error messages when there's a misconfiguration, and require people to set up their kubectl's appropriately

I think I'd like to see this command as the first entry-point for users new to a cluster.  They're trying to do something on the cluster, they don't want to control their .kube/config directly (that's really hard), so they want a single command that sets up their server stanza and their user stanza in one shot.

I think you can use the `auth.Login()` to satisfy the user stanza, but not the server stanza.

@cjcullen This is an example of where in the general case, we need to have an stdin, stdout, stderr plumbed into `Login`.  The next logical flow to have here (maybe even the first logical flow) is one that handles basic auth against your cluster.